util/parsenum.h

```diff
@@ -18,13 +18,13 @@
 #define PARSENUM(x, s, min, max)					\
 	(								\
 		errno = 0,						\
-		(((x) = 1, (x) /= 2) != 0) ?				\
-			((x) = parsenum_float((s), (min), (max))) :	\
-		(((x) = -1) <= 0) ?					\
-			((x) = parsenum_signed((s), (min), (max))) :	\
+		(((*x) = 1, (*x) /= 2) != 0) ?				\
```

Comment:
This is broken.  You need to replace `(x)` with `(*(x))` in order to avoid problems if `x` is something like `&foo + 5`.